routes/delegate/register.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response } from 'express';
+import * as bip39 from 'bip39';
+import { HDKey } from 'ethereum-cryptography/hdkey';
+import crypto from 'crypto';
+import { getProvider, mintCapacityCredits, sendLitTokens } from '../../lit';
+import { ethers, type Wallet } from 'ethers';
+
+// Takes an arbitrary string, and converts it deterministically to a number
+//
+// FIXME: This is a naive implementation, and could result in collisions
+//        in the future. We should consider using a formalized api-key
+//        that contains a unique identifier for the user.
+export function normalizeApiKey(apiKey: string): number {
+    const hash = crypto
+        .createHash('sha256')
+        .update(apiKey)
+        .digest('hex');
+
+    return parseInt(hash, 16);
+}
+
+export function generateApiSecret(): string {
+    return crypto.randomBytes(64).toString('base64');
+}
+
+export async function deriveWallet(apiKey: string, apiSecret: string) {
+    const mnemonic = process.env.LIT_DELEGATION_ROOT_MNEMONIC;
+
+    if (!mnemonic) {
+        throw new Error("Mnemonic not set");
+    }
+
+    console.log(`Deriving wallet for ${apiKey} with secret ${apiSecret}`)
+
+    const seed = await bip39.mnemonicToSeed(mnemonic);
```

Comment:
Can we use ethers, which is already imported and included, for the derivation?  https://docs.ethers.org/v5/api/utils/hdnode/

Seems like the implementation is basically the same?

Reply:
Done